### PR TITLE
[r/python] Deprecation notices for `used_shape`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** For #2407

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

File: apis/python/src/tiledbsoma/_types.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
     NPInteger = np.integer[npt.NBitBase]
     NPFloating = np.floating[npt.NBitBase]
     NPNDArray = npt.NDArray[np.number[npt.NBitBase]]
-    PDSeries = pd.Series[Any]  # type: ignore[misc] # a pd.Series of "any" type will raise mypy error
+    # A pd.Series of "Any" type will raise mypy error:
```
Comment (johnkerl) [inline apis/python/src/tiledbsoma/_types.py:23]: Not related to this PR. Lint flared up today right here. IDK why.

**Review by mojaveazure — APPROVED** [review]

**Comment by mojaveazure** [comment]
For the R API, you could also call `.Deprecated()` in `used_shape()` to throw a deprecation warning when calling that method

**Comment by johnkerl** [comment]
We have some flakiness around (I think) the particular linter version available -- something bumped today -- lint passing on Python 3.8 and failing on Python 3.11, and it appears they cannot be both appeased simultaneously ...

**Comment by johnkerl** [comment]
> For the R API, you could also call .Deprecated() in used_shape() to throw a deprecation warning when calling that method

Oops @mojaveazure I meant to do that on this PR -- I'll put up another

**Pull request merged**